subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildIntegrationTest.groovy

```diff
@@ -995,6 +995,7 @@ task generate(type: TransformerTask) {
 """
 
         when:
+        executer.expectDeprecationWarning(":transform1 consumes the output of :src2, but does not declare a dependency. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0.")
```

Comment:
I don't think the deprecation should be a failure in 7.0. Though the warning message will have 8.0 in it very soon.

Reply:
Is there a reason not to change it to `willBeRemovedInGradle8()` instead of `willBeRemovedInGradle7()` already?

Reply:
Yes, because it now runs through our validation infrastructure, so I can't change the `willBeRemovedInGradle7()` easily, since I would need to add a new flag for task validations.